service/status-report/main.go

```diff
@@ -202,3 +144,79 @@ func main() {
 		klog.Exitf("Failed to update mon configmap for storageClient %v: %v", storageClient.Status.ConsumerID, err)
 	}
 }
+
+func setClusterInformation(ctx context.Context, cl client.Client, status interfaces.StorageClientStatus) {
+	var clusterID configv1.ClusterID
+	clusterVersion := &configv1.ClusterVersion{}
+	clusterVersion.Name = "version"
+	if err := cl.Get(ctx, types.NamespacedName{Name: clusterVersion.Name}, clusterVersion); err != nil {
+		klog.Warningf("Failed to get clusterVersion: %v", err)
+	} else {
+		clusterID = clusterVersion.Spec.ClusterID
+	}
+
+	clusterConfig := &corev1.ConfigMap{}
+	clusterConfig.Name = clusterConfigName
+	clusterConfig.Namespace = clusterConfigNamespace
+
+	if err := cl.Get(ctx, client.ObjectKeyFromObject(clusterConfig), clusterConfig); err != nil {
+		klog.Warningf("Failed to get clusterConfig %q/%q: %v", clusterConfig.Namespace, clusterConfig.Name, err)
+	}
+
+	clusterMetadataYAML := clusterConfig.Data["install-config"]
+	clusterMetadata := struct {
+		Metadata struct {
+			Name string `yaml:"name"`
+		} `yaml:"metadata"`
+	}{}
+	err := yaml.Unmarshal([]byte(clusterMetadataYAML), &clusterMetadata)
+	if err != nil {
+		klog.Warningf("Fatal error, %v", err)
+	}
+	clusterName := ""
+	if len(clusterMetadata.Metadata.Name) > 0 {
+		clusterName = clusterMetadata.Metadata.Name
```

Comment:
following similar approach in this file, use the condition to only log the warning.

Reply:
not changed, I meant
```
if empty {
 log a warning that name is not found
}
```

Reply:
@raaizik could you pls flip the condition as I mentioned in my comment and I'd be able to merge. 

From
```
var name string
if not empty {
  set name from meta
} else {
  log warning
}
setStatus(name)
```

To
```
If empty {
  log warning
}
setStatus(name from meta) // no need for extra var
```

Reply:
Your suggestion doesn't capture the case of a nil `clusterMetadata.Metadata` while mine does.

Reply:
why would `clusterMetadata.Metadata` be nil as I see no field in the struct (or struct itself) will have a zero value of `nil`? Not related, just a point, if `clusterMetadata.Metadata` can be `nil` then existing `len(clusterMetadata.Metadata.Name)` would panic already.

I'm not able to see how my suggestion will fail known cases or it is incorrect, let me know otherwise, thanks.

``` go
package main

import (
	"fmt"

	"k8s.io/apimachinery/pkg/util/yaml"
)

type meta struct {
	Metadata struct {
		Name string `yaml:"name"`
	} `yaml:"metadata"`
}

var noData string

var noMeta = `
---
apiVersion: v1
controlPlane:
  replicas: 1
networking:
  machineNetwork:
platform:
  none: {}
`

var noName = `
---
additionalTrustBundlePolicy: Proxyonly
apiVersion: v1
baseDomain: example.com
metadata:
  creationTimestamp: null
networking:
  clusterNetwork:
  - cidr: 10.0.0.0/14
    hostPrefix: 23
`

var withName = `
---
additionalTrustBundlePolicy: Proxyonly
apiVersion: v1
baseDomain: example.com
metadata:
  creationTimestamp: null
  name: clusterName
networking:
  clusterNetwork:
  - cidr: 10.0.0.0/14
    hostPrefix: 23
`

func main() {
	var err error

	noData_ := &meta{}
	if err = yaml.Unmarshal([]byte(noData), noData_); err != nil {
		panic("failed unmarshal")
	}
	fmt.Printf("noData: %q\n", noData_.Metadata.Name)

	noMeta_ := &meta{}
	if err = yaml.Unmarshal([]byte(noMeta), noMeta_); err != nil {
		panic("failed unmarshal")
	}
	fmt.Printf("noMeta: %q\n", noMeta_.Metadata.Name)

	noName_ := &meta{}
	if err = yaml.Unmarshal([]byte(noName), noName_); err != nil {
		panic("failed unmarshal")
	}
	fmt.Printf("noName: %q\n", noName_.Metadata.Name)

	withName_ := &meta{}
	if err = yaml.Unmarshal([]byte(withName), withName_); err != nil {
		panic("failed unmarshal")
	}
	fmt.Printf("withName: %q\n", withName_.Metadata.Name)

	// $ go run main.go
	// noData: ""
	// noMeta: ""
	// noName: ""
	// withName: "clusterName"
}
```

Reply:
> why would

Not about why, it's about what if. 

> would panic already

And that is my concern, that the flow would resume to the line of setting the status even if, for example, `Name` is nil (condition doesn't panic in this case).
```go
	clusterName := ""
	if len(clusterMetadata.Metadata.Name) == 0 {
		klog.Warningf("Cluster name is empty, %v", err)
	}
	status.SetClusterName(clusterMetadata.Metadata.Name)
```

Reply:
> it's about what if.
- still, why would the zero value of a string (here `clusterMetadataYAML`) be `nil` and none of the fields of `clusterMetadata` struct can have a zero value of `nil` the field types are `strings` only?

Reply:
@raaizik @leelavg 
`Name` is of type `string` not `*string` so it cannot accept a `nil` value only an empty string value. That means you will never get to a point where accessing `.Name` will panic.  The only question left is what should we do in case we have an empty string for a name but I think that is already answered above with the discussion about having an warning at that case

Reply:
Changed

Reply:
as discussed offline, I hope below helps
``` go
package main

import (
	"fmt"

	"k8s.io/apimachinery/pkg/util/yaml"
)

type pointerMeta struct {
	Metadata *struct {                // here the type is a pointer
		Name string `yaml:"name"`
	} `yaml:"metadata"`
}

type valueMeta struct {
	Metadata struct {                // here the type is a struct
		Name string `yaml:"name"`
	} `yaml:"metadata"`
}

var noMeta = `
---
apiVersion: v1
controlPlane:
  replicas: 1
networking:
  machineNetwork:
platform:
  none: {}
`

func main() {
	var err error

	noMetaValue_ := &valueMeta{}
	if err = yaml.Unmarshal([]byte(noMeta), noMetaValue_); err != nil {
		panic("failed unmarshal")
	}
	fmt.Printf("noMetaValue_: %q\n\n", noMetaValue_.Metadata.Name)

	noMetaPtr_ := &pointerMeta{}
	if err = yaml.Unmarshal([]byte(noMeta), noMetaPtr_); err != nil {
		panic("failed unmarshal")
	}
	fmt.Printf("noMetaPtr_: %q\n", noMetaPtr_.Metadata.Name)

	// $ go run main.go
	// noMetaValue_: ""

	// panic: runtime error: invalid memory address or nil pointer dereference
	// [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4fbf78]

	// goroutine 1 [running]:
	// main.main()
	//         /tmp/meta/main.go:45 +0xf8
	// exit status 2
}
```